src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java

```diff
@@ -461,6 +464,29 @@ public void startCaseBookPDFJob(JobDetail jobDetail,
 			    for(StudyEvent studyEvent : subjectStudyEvents) {
 			    	List<EventCrf> eventCRFs = studyEvent.getEventCrfs();
 			    	
+			    	/*
+			    	 * OC-11782
+			    	 * check the CRF ordinal and reorder by the ordinal 
+			    	 */
+			    	List<EventDefinitionCrf> edfcs = studyEvent.getStudyEventDefinition().getEventDefinitionCrfs();
+			    	for(EventCrf eventCrf : eventCRFs) {
+			    		for(EventDefinitionCrf edfc:edfcs) {
+			    			CrfBean crf = edfc.getCrf();
+			    			
+			    			if(eventCrf.getCrfVersion().getCrf().getCrfId() == crf.getCrfId()) {
+			    				eventCrf.setOrdinal(edfc.getOrdinal());
+			    			}
+			    		}			    		
+			    	}
+			    	
+			    	Comparator<EventCrf> compareByOrinal = new Comparator<EventCrf>() {
```

Comment:
make this a static variable and move this comparator to EventCrf.java. Update the comparison logic such that you can fetch the ordinal directly from EvenCrf. 
Please check with Joe to find out if we should use FormLayout or CrfVersion

Reply:
done